src/main/java/knightminer/inspirations/plugins/jei/smashing/SmashingItemRecipeWrapper.java

```diff
@@ -0,0 +1,74 @@
+package knightminer.inspirations.plugins.jei.smashing;
+
+import java.awt.*;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import knightminer.inspirations.library.Util;
+import knightminer.inspirations.library.recipe.anvil.ISimpleAnvilRecipe;
+import mezz.jei.api.ingredients.IIngredients;
+import mezz.jei.api.recipe.IRecipeWrapper;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.Minecraft;
+import net.minecraft.item.ItemStack;
+
+public class SmashingItemRecipeWrapper implements IRecipeWrapper {
+
+	protected final ISimpleAnvilRecipe recipe;
+	private final List<List<ItemStack>> input;
+	private final ImmutableList<ItemStack> output;
+
+	private String heightRequirementString;
+	private String blockRequirementString;
+
+	public SmashingItemRecipeWrapper(ISimpleAnvilRecipe recipe) {
+		this.recipe = recipe;
+		this.input = this.recipe.getInput().stream().map(ImmutableList::of).collect(Collectors.toList());
+		this.output = ImmutableList.copyOf(this.recipe.getResult());
+
+		Integer minFallHeight = recipe.getFallHeight();
+		this.heightRequirementString =
+				minFallHeight != null ? Util.translateFormatted("gui.jei.anvil_smashing.height", minFallHeight) : null;
+
+		Object inputState = recipe.getInputState();
+		this.blockRequirementString = inputState instanceof IBlockState ?
```

Comment:
Might be nice to have this show visually in JEI, maybe place this under the item that is under the anvil? Or alternatively, what about if we place the items on top of the anvil and the blocks underneath, to make it extra clear which are items